scipy/special/tests/test_basic.py

```diff
@@ -1545,6 +1545,36 @@ def test_errprint(self):
         assert_equal(d,b)  # makes sure state was returned
         # assert_equal(d,1-a)
 
+    def test_erf_nan_inf(self):
+        vals = [np.nan, -np.inf, np.inf]
```

Comment:
That's not quite enough, you'd also need to customize the error message to make clear what failed. And I'd rather have all the tests run instead of stopping on the first failure. So the alternative here is a test generator (which I tend to avoid unless the gain in LoC is really large).